### PR TITLE
Fix purs bundle to recognise explicit paths

### LINKED ISSUE #3359: `purs bundle` doesn't bundle anything
I think this is perhaps a knock-on effect from the change to include `index.js` in require paths.

### PR DESCRIPTION
**Pull request by garyb**
Resolves #3359 

Could do with a test for bundling so this doesn't happen again, but I'm not sure how best to achieve that really.

@chexxor sorry, I know you said you were going to take a look, but I knew how to do this pretty easily so thought I'd just get it in :smile:

### CODE REVIEW

File: src/Language/PureScript/Bundle.hs [hunk 1]
```diff
@@ -145,13 +145,20 @@ printErrorMessage (MissingMainModule mName) =
...
-checkImportPath "./foreign" m _ =
+checkImportPath "./foreign.js" m _ =
   Right (ModuleIdentifier (moduleName m) Foreign)
 checkImportPath name _ names
-  | Just name' <- stripPrefix "../" name
+  | Just name' <- stripSuffix "/index.js" =<< stripPrefix "../" name
   , name' `S.member` names = Right (ModuleIdentifier name' Regular)
 checkImportPath name _ _ = Left name
 
+stripSuffix :: Eq a => [a] -> [a] -> Maybe [a]
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Bundle.hs]: Could this just be: https://hackage.haskell.org/package/filepath-1.4.2/docs/System-FilePath-Posix.html#v:takeDirectory?

Comment (kRITZCREEK) [reply src/Language/PureScript/Bundle.hs]: I guess it doesn't do the `Maybe` thing...

Comment (garyb) [reply src/Language/PureScript/Bundle.hs]: Perhaps, yeah - but this way at least ensures it's exactly the expected pattern, whereas that might be too broad, unless we check the filename part separately too.

**Comment by garyb** [comment]
Ah sorry, thanks for looking into it and working on it anyway!

**Pull request merged**